tuxedo-core/src/executive.rs

```diff
@@ -291,8 +303,25 @@ impl<
         // be cleared before the end of the block
         sp_io::storage::set(HEADER_KEY, &block.header().encode());
 
+        // Tuxedo requires that inherents are at the beginning (and soon end) of the
+        // block and not scattered throughout. We use this flag to enforce that.
+        let mut finished_with_opening_inherents = false;
+
         // Apply each extrinsic
         for extrinsic in block.extrinsics() {
+            // Enforce that inherents are in the right place
+            match (
```

Comment:
you might compress `!a && b` and `a && !b` using a XOR, since both result into  `=> (), `

or you might go hard and do (untested)
```rust
if a == b {
    if a {
        panic!("Tried to execute opening inherent after switching to non-inherents.");
    }
    finished_with_opening_inherents = true, 
}
```

I think the match is more readable 😄

Reply:
Okay, you guys inspired me and I revised this in 87ee4dca4a517b418457195e86053a59ca7b5195

Reply:
I'm not surprised that there was a simpler solution. I'm tired lol